dev-packages/ovsx-client/src/ovsx-http-client.ts

```diff
@@ -43,10 +47,20 @@ export class OVSXHttpClient implements OVSXClient {
     }
 
     protected async requestJson<R>(url: string): Promise<R> {
-        return RequestContext.asJson<R>(await this.requestService.request({
-            url,
-            headers: { 'Accept': 'application/json' }
-        }));
+        for (let i = 0; i < 10; i++) {
+            await this.rateLimiter.removeTokens(1);
+            const context = await this.requestService.request({
+                url,
+                headers: { 'Accept': 'application/json' }
+            });
+            if (context.res.statusCode === 429) {
```

Comment:
Shouldn't we at least output some diagnostic in this case? Can we determine something from the response, like "retry-after" or similar?

Reply:
That'd be great. Unfortunately, Open-VSX isn't really helpful:
```ts
{
  server: 'nginx',
  date: 'Mon, 12 Aug 2024 18:49:43 GMT',
  'content-type': 'application/json',
  'content-length': '35',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-rate-limit-retry-after-seconds': '0', // <-- ???
  'access-control-expose-headers': 'X-Rate-Limit-Retry-After-Seconds, X-Rate-Limit-Remaining',
  'access-control-allow-origin': '*'
}
```

Reply:
😞